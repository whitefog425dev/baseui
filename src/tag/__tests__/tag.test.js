/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import React from 'react';
import {mount} from 'enzyme';
import {Tag, StyledAction} from '../index.js';

describe('Stateless tag', function() {
  let wrapper,
    events = {},
    children;
  let allProps: any = {},
    mockFn;

  beforeEach(function() {
    mockFn = jest.fn();
    events = {
      onActionClick: mockFn,
    };
    allProps = {
      ...events,
    };
    children = 'Some tag';
  });

  afterEach(function() {
    jest.restoreAllMocks();
    wrapper && wrapper.unmount();
  });

  test('should render component', function() {
    wrapper = mount(<Tag {...allProps}>{children}</Tag>);
    expect(wrapper).toMatchSnapshot('Component has correct render');
  });

  describe('On action', function() {
    beforeEach(function() {
      allProps.onActionClick = jest.fn();
      wrapper = mount(<Tag {...allProps}>{children}</Tag>);
    });

    test('should call callback if action button is clicked', function() {
      const actionIcon = wrapper.find(StyledAction);
      actionIcon.first().simulate('click');
      expect(allProps.onActionClick).toHaveBeenCalledWith(
        allProps.onActionClick.mock.calls[0][0],
        children,
      );
    });
  });
});
