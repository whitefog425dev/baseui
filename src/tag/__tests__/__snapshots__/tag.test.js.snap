// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stateless tag should render component: Component has correct render 1`] = `
<Tag
  disabled={false}
  isFocused={false}
  isHovered={false}
  kind="primary"
  onActionClick={[MockFunction]}
  overrides={Object {}}
>
  <MockStyledComponent
    $disabled={false}
    $isFocused={false}
    $isHovered={false}
    $kind="primary"
  >
    <span
      styled-component="true"
    >
      Some tag
      <MockStyledComponent
        $disabled={false}
        $isFocused={false}
        $isHovered={false}
        $kind="primary"
        onClick={[Function]}
      >
        <div
          onClick={[Function]}
          styled-component="true"
        >
          <MockStyledComponent
            fill="none"
            height="8"
            viewBox="0 0 8 8"
            width="8"
            xmlns="http://www.w3.org/2000/svg"
          >
            <svg
              fill="none"
              height="8"
              styled-component="true"
              viewBox="0 0 8 8"
              width="8"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M0.861278 0.862254C1.12163 0.601905 1.54374 0.601905 1.80409 0.862254L3.99935 3.05752L6.19461 0.862254C6.45496 0.601905 6.87707 0.601905 7.13742 0.862254C7.39777 1.1226 7.39777 1.54471 7.13742 1.80506L4.94216 4.00033L7.13742 6.19559C7.39777 6.45594 7.39777 6.87805 7.13742 7.1384C6.87707 7.39875 6.45496 7.39875 6.19461 7.1384L3.99935 4.94313L1.80409 7.1384C1.54374 7.39875 1.12163 7.39875 0.861278 7.1384C0.600928 6.87805 0.600928 6.45594 0.861278 6.19559L3.05654 4.00033L0.861278 1.80506C0.600928 1.54471 0.600928 1.1226 0.861278 0.862254Z"
                fill="currentColor"
                fillRule="evenodd"
              />
            </svg>
          </MockStyledComponent>
        </div>
      </MockStyledComponent>
    </span>
  </MockStyledComponent>
</Tag>
`;
